src/uct/sm/mm/base/mm_ep.c

```diff
@@ -236,14 +236,33 @@ static inline void uct_mm_ep_update_cached_tail(uct_mm_ep_t *ep)
     ep->cached_tail = ep->fifo_ctl->tail;
 }
 
+static UCS_F_ALWAYS_INLINE void uct_mm_ep_peer_check(uct_mm_ep_t *ep,
+                                                     unsigned flags)
+{
+    ucs_status_t status;
+
+    if (ucs_unlikely(flags & UCT_SEND_FLAG_PEER_CHECK)) {
+        status = uct_ep_check(&ep->super.super, 0, NULL);
+        ucs_assertv(status == UCS_OK, "%s", ucs_status_string(status));
```

Comment:
can we call uct_ep_keepalive_check directly? then no need assert

Reply:
good idea, done